plugins/modules/inventory_intent.py

```diff
@@ -1151,6 +1216,199 @@ def provisioned_wired_device(self):
+            response = self.dnac_apply['exec'](
+                family="sites",
+                function='get_site',
+                params={"name": site_name},
+            )
+
+            if not response:
+                self.msg = "Site - {0} not found".format(site_name)
+                self.log(self.msg)
+                return site_type
+
+            self.log(str(response))
+            site = response.get("response")
+            site_additional_info = site[0].get("additionalInfo")
+
+            for item in site_additional_info:
+                if item["nameSpace"] == "Location":
+                    site_type = item.get("attributes").get("type")
+
+        except Exception:
+            self.module.fail_json(msg="Site not found", response=[])
+
+        return site_type
+
+    def provisioned_wireless_devices(self, device_ips):
+        """
+        Provision Wireless devices in Cisco DNA Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco DNA Center.
+            device_ips (list): List of IP addresses of the devices to be provisioned.
+        Returns:
+            self (object): An instance of the class with updated result, status, and log.
+        Description:
+            This function performs wireless provisioning for the provided list of device IP addresses.
+            It iterates through each device, retrieves provisioning parameters using the get_wireless_param function,
+            and then calls the Cisco DNA Center API for wireless provisioning. If all devices are already provisioned,
+            it returns success with a relevant message.
+        """
+
+        provision_count, already_provision_count = 0, 0
+
+        for device_ip in device_ips:
+            try:
+                # Collect the device parameters from the playbook to perform wireless provisioing
+                provisioning_params = self.get_wireless_param(device_ip)
+
+                # Now we have provisioning_param so we can do wireless provisioning
+                response = self.dnac_apply['exec'](
+                    family="wireless",
+                    function="provision",
+                    op_modifies=True,
+                    params=provisioning_params,
+                )
+
+                if response.get("status") == "failed":
+                    description = response.get("description")
+                    error_msg = "Cannot do Provisioning for Wireless device {0} beacuse of {1}".format(device_ip, description)
+                    self.log(error_msg)
+                    continue
+
+                task_id = response.get("taskId")
+
+                while True:
+                    execution_details = self.get_task_details(task_id)
+                    progress = execution_details.get("progress")
+                    self.log(progress)
+
+                    if 'TASK_PROVISION' in progress:
+                        self.status = "success"
```

Comment:
As there are lot of self.status, can we simplify it? 



```
                if 'TASK_PROVISION' in progress:
                    self.handle_successful_provisioning(device_ip, execution_details)
                    provision_count += 1
                    break
                elif execution_details.get("isError"):
                    self.handle_failed_provisioning(device_ip, execution_details)
                    break
..
..
..

def handle_successful_provisioning(self, device_ip, execution_details):
    self.status = "success"
    self.result['changed'] = True
    self.result['response'] = execution_details
    self.log(f"Wireless Device {device_ip} provisioned successfully!!")

def handle_failed_provisioning(self, device_ip, execution_details):
    self.status = "failed"
    failure_reason = execution_details.get("failureReason", "Unknown failure reason")
    self.msg = f"Wireless Device Provisioning failed for {device_ip} because of {failure_reason}"

```

Reply:
added new api and changed in both provisioned_wired_device and provisioned_wireless_device API's.